### PR TITLE
added match statement instead of if-else block

### PR DESCRIPTION
**Pull request by SmartManLudo**
# Objective
Replaced the if-else statement block in favor of a match statement to give a more cleaner code.

### CODE REVIEW

**Review by cart — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@SmartManLudo CI is failing due to formatting issues. Like it says, can you run `cargo fmt --all` and then push your changes so this can get merged?

File: crates/bevy_sprite/src/mesh2d/mesh.rs [hunk 1]
```diff
@@ -232,12 +232,7 @@ impl FromWorld for Mesh2dPipeline {
                 &image.data,
                 ImageDataLayout {
                     offset: 0,
-                    bytes_per_row: Some(
-                        std::num::NonZeroU32::new(
-                            image.texture_descriptor.size.width * format_size as u32,
-                        )
-                        .unwrap(),
-                    ),
+                    bytes_per_row: Some(image.texture_descriptor.size.width * format_size as u32),
```
Comment (mockersf) [inline crates/bevy_sprite/src/mesh2d/mesh.rs:235]: I'm not sure why this change is in this PR... this has already been done on `main`, and either your branch is not up to date with `main` and you shouldn't need it anyway as it's from the `wgpu` update, or it should already be done

**Comment by alice-i-cecile** [comment]
@SmartManLudo can you address the comment from @mockersf above? Rebasing may work to fix it.

File: crates/bevy_sprite/src/mesh2d/mesh.rs (hunk 1 above)
Comment (SmartManLudo) [reply crates/bevy_sprite/src/mesh2d/mesh.rs]: I haven't a clue why this happened too. Should be fixed now...
